material_components_ios.bzl

```diff
@@ -0,0 +1,49 @@
+"""Baze macros for building MDC component libraries."""
+
+load("@bazel_ios_warnings//:strict_warnings_objc_library.bzl", "strict_warnings_objc_library")
+
+def mdc_objc_library(
+    name,
+    copts = [],
+    **kwargs):
+  """Declare an Objective-C library with MDC's common warning flags."""
+  strict_warnings_objc_library(
+      name = name,
+      copts = copts,
+      **kwargs)
+
+def mdc_public_objc_library(
+    name,
+    deps = [],
+    extra_srcs = [],
+    sdk_frameworks = [],
+    visibility = ["//visibility:public"],
+    **kwargs):
+  """Declare a public MDC component as a Objective-C library according to MDC's conventions.
+
+  The conventions for an MDC component are:
+  - The public implementation lives inside of src.
+  - The private implementation lives inside of src/private.
+  - Headers are imported relative to src.
+
+  The default visibility of "//visibility:public" can be overridden, for
+  example, "//some/package:__subpackages__".
+
+  Args:
+    name: The name of the library.
+    deps: The dependencies of the library.
+    extra_srcs: Extra sources to add to the standard ones.
+    sdk_frameworks: The SDK frameworks needed, e.g. "CoreGraphics".
+    visibility: The visibility of the package.
+    **kwargs: Any arguments accepted by _mdc_objc_library().
+  """
+  mdc_objc_library(
+      name = name,
+      deps = deps,
+      sdk_frameworks = sdk_frameworks,
+      visibility = visibility,
+      srcs = native.glob(["src/*.m", "src/private/*.h", "src/private/*.m"]) + extra_srcs,
```

Comment:
Should the `src/private/*.h` glob move down to `hdrs`?

Reply:
Nope - `hdrs` defines the files that are allowed to be imported by a client and we don't want clients pulling in private headers.

> `hdrs`: A list of header files. These are the header files that are exported by this rule.

https://docs.bazel.build/versions/master/be/c-cpp.html#cc_inc_library.hdrs

Reply:
Ack. I think this is the link, though:

> The list of C, C++, Objective-C, and Objective-C++ files that are included as headers by source files in this rule or by users of this library. These will be compiled separately from the source if modules are enabled.

https://docs.bazel.build/versions/master/be/objective-c.html#objc_library.hdrs